### PR TITLE
New layout

### PR DESCRIPTION
**Pull request by smorloc**
Let's try this one more time...

### CODE REVIEW

**Comment by Intrinsically-Sublime** [comment]
Please do not make this the new interface. Most of the input fields are below the bottom of the screen on small monitors which make it unusable, it needs to be two columns wide. Also if you double click the panel divider it makes the 3d view disappear without any way to bring it back except restarting Cura

**Comment by smorloc** [comment]
The new layout is my bad.  On many screens, even most laptops, I found
there was a substantial amount of unused space below the input fields while
the two columns took away from the 3D view.

Could you tell me what resolution you are running at?

I will look into the double clicking on the panel divider.  Frankly, it
never dawned on my to test that...

**Comment by Intrinsically-Sublime** [comment]
On all of my Laptops it causes an issue. Right now I am on a brand new Samsung 15.6" 1366 x 768 and if the window is maximised from top to bottom some of the settings on the advanced tab are still hidden and I have to press alt+f7 to move the window up to see them. On my old 14" acer it is even worse. It may be that I am running Debian (testing)(wheezy) and Linux Mint 12 so the fonts may be the difference. Either way I am sure it will cause even more issues on machines with lower resolutions and different distros. Also most monitors nowadays are 16x9 so they have more real estate horizontally than vertically. If the purpose is to give the 3D view more room maybe having it on its own tab would be best. It is not like you really need to see it when changing settings.

**Comment by smorloc** [comment]
The double clicked on the sash behavior was clearly a bug and I have
implemented a fix for that.

I see that if one shrunk the window vertically, the settings were cut off
without a way of getting to them.  This was also a bug and I address it
with scrollbars that are dynamically displayed when needed.

I don't under your statment about hitting 'alt'-f7'.  If the window is
already maximized, how did that key sequence help?
I typically test the Unix version using Ubuntu/Gnome.  'alt-f7' doesn't
seem like it does anything when I try it.  I haven't looked through the
Ubuntu/Gnome docs though...

FYI, the following is what the display looks like on a 1280x768
Ubuntu/Gnome display.  Would it be possible to see a screen shot on your
system?

Screenshot from 2013-01-23 23 50 15

**Comment by Intrinsically-Sublime** [comment]
It was only half maximized (drag the window to one edge and it will fill half the screen) or stretch it to fit from top to bottom. Then when you press alt+f7 in Gnome 3 it allows you to move the window without grabbing the top bar since it will be outside the desktop after moving it. Then you can grab the bottom corner and expand the window even further to see the rest of the settings.

Your screen shot did not show up.

Also I don't believe a scroll bar is the answer. Having a scroll bar is just an inconvenience and your changes do not serve any purpose other than making it more pleasing to you. To me I want to see my settings as they are the most important part of the slicing software.

Screenshot from 2013-01-23 21:14:26

**Comment by Intrinsically-Sublime** [comment]
After doing the whole alt+f7 thing and expanding the window and then bringing it back down to see the top bar the load model and prepare print buttons are off the screen.

Note this is on the 15.6" 1366 x 768 Samsung Laptop with Debian Wheezy Gnome 3 which is almost completely a default install.
Screenshot from 2013-01-23 21:21:08

**Comment by smorloc** [comment]
It looks like the only parameters you are not seeing is 'Duplicate Outlines'.  Out of curiosity, do you see it if you maximize the window (not enlarge it by hand and move it around)?

Sorry about the missing image.  I can't figure out why it disappeared.  Let me try again...  

The following two images are: the new code on my 1280x768 display and the currently released Cura 12.12 on a 1280x768 display.  The reason why I tried to come up with the alternative layout was that the Cura 12.12 display was dominated by whitespace.

Screenshot from 2013-01-23 23 50 15

Screenshot from 2013-01-24 00 22 57

**Comment by Intrinsically-Sublime** [comment]
Yes but why would you ever have Cura maximized? It makes it hard to drag items onto it and really you never have it as your main focus on the screen. It is always something you start and go away or go back to whatever else you are doing on your computer. Its not like a Browser or text editor or something where you need to use as much of the screen as possible. For me it looks like the following picture 100% of the time and I will never use it full screen.

Old way with all setting visible and my file browser and my open firefox all accessible.
Screenshot from 2013-01-23 21:45:23

New way almost no settings exposed.
Screenshot from 2013-01-23 21:45:15

Maximized with the window frame removed it takes up the entire screen without access to drag items into the build area. And makes for lots of Blue space.
Screenshot from 2013-01-23 21:48:43

**Comment by Intrinsically-Sublime** [comment]
How hard would it be to automatically go from two columns when windowed to one column when maximized?

**Comment by Intrinsically-Sublime** [comment]
Further problem is even when you have it maximized you start to lose settings after you slice and the bottom bar starts to cover them. With the old way you could slice a bunch of stuff before it started to intrude on the settings.

**Comment by daid** [comment]
Ah, I knew this change would spark some discussion :-)

I think the switch between two/one column idea won't be a bad thing. I also have a lot of wasted screen space on full screen. But I liked the two-column better for window mode. (I did design it to fit on small resolution screens, that's why it was 2 columns in the first place)

**Comment by smorloc** [comment]
Rest had cleared my head... thank you Daid/Sublime for your patience...

How about if the application is changed to switch to two column mode when
you widen the left hand pane enough to accommodate two columns? If you
narrow it up, it will switch to single column with a scroll bar (if the
window is too short to accommodate all the options). That way the user has
control over the number of columns regardless of screen size.  We could
also change the default sash/slider position so that two columns is the
default...

**Comment by Intrinsically-Sublime** [comment]
That sounds like a good plan, it will fit every users preferences.

**Comment by smorloc** [comment]
I am actively working on this.  The basic behavior is implemented however I
am trying to streamline the hairy layout calls and test them across the 3
primary platforms (Windows, Mac OS, Linux).

**Comment by daid** [comment]
Getting the layouts working on all 3 platforms is really annoying. I thought wxWidgets would solve that for you, but in quite a few cases Windows and Mac act differently. (Usually when windows works Linux also works)

But good that you are on it Steve, allows me to concentrate on the new OpenGL buttons (yes, I know they are butt ugly right now) and getting the project planner working again.
